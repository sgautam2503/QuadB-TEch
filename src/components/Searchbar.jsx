import PropTypes from 'prop-types';
import '../styles/index.css'

import { BsSearch } from "react-icons/bs";

function Searchbar({ showName, handleSearchShow }) {
  return (
    <div className='search'>
    <input className='input'
      type="search"
      placeholder=" Search what's in your heart..."
      aria-label="Search"
      value={showName}
      onChange={handleSearchShow}
    />
    </div>
  );
}

Searchbar.propTypes = {
  showName: PropTypes.string,
  handleSearchShow: PropTypes.func,
}.isRequired;

export default Searchbar;
