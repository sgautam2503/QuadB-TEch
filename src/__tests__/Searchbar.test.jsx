import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Searchbar from '../components/Searchbar';
import store from '../redux/Store';

describe('Searchbar component', () => {
  it('should display on the screen', () => {
    const searchBarComponent = render(
      <BrowserRouter>
        <Provider store={store}>
          <Searchbar />
        </Provider>
      </BrowserRouter>,
    );

    const searchBar = screen.getByRole('searchbox');
    screen.debug(searchBar);
    expect(searchBar).toBeInTheDocument();
    expect(searchBarComponent).toMatchSnapshot();
  });
});
