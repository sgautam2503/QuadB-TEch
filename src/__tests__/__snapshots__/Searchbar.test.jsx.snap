// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Searchbar component > should display 2 shows on the screen 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card bg-primary text-light"
        style="height: 8vh;"
      >
        <input
          aria-label="Search"
          class="p-3"
          placeholder="Search a show ..."
          type="search"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card bg-primary text-light"
      style="height: 8vh;"
    >
      <input
        aria-label="Search"
        class="p-3"
        placeholder="Search a show ..."
        type="search"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Searchbar component > should display on the screen 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card bg-primary text-light"
        style="height: 8vh;"
      >
        <input
          aria-label="Search"
          class="p-3"
          placeholder="Search a show ..."
          type="search"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card bg-primary text-light"
      style="height: 8vh;"
    >
      <input
        aria-label="Search"
        class="p-3"
        placeholder="Search a show ..."
        type="search"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
